### PR TITLE
bug: owner tokens not updated on token transfer

### LINKED ISSUE #3: _ownerTokens not updated on token transfer

### PR DESCRIPTION
**Pull request by jmulq**
Fixes #3.

Add solhint and prettier for solidity.

Add unit tests to source control.

### CODE REVIEW

File: contracts/VLAND.sol [hunk 1]
```diff
@@ -1,42 +1,61 @@
...
+
     uint256 private _maxSupply;
 
-    function VLAND_init(address admin, string memory name_, string memory symbol_, string memory baseUri_, uint256 maxSupply_) initializer public {
+    function initialize(
+        address admin,
+        string memory name_,
+        string memory symbol_,
+        string memory baseUri_,
+        uint256 maxSupply_
```
Comment (UmairJanjua6) [inline contracts/VLAND.sol:30]: As we have fix total supply, so it's better to define it at storage level

File: contracts/VLAND.sol [hunk 2]
```diff
@@ -1,42 +1,61 @@
...
+    ) public initializer {
         __ERC721_init(name_, symbol_);
+        __ERC721Enumerable_init();
         _baseURIextended = baseUri_;
         _maxSupply = maxSupply_;
         _setupRole(DEFAULT_ADMIN_ROLE, admin);
     }
 
-    function maxSupply() external view returns(uint256) {
+    function maxSupply() external view returns (uint256) {
```
Comment (UmairJanjua6) [inline contracts/VLAND.sol:39]: renaming maxSupply to totalSupply is better as many new platforms are only reading the totalSupply function now

File: contracts/VLAND.sol [hunk 3]
```diff
@@ -1,42 +1,61 @@
...
+        uint256[] memory tokenIds = new uint256[](balance);
+
+        for (uint256 index = 0; index < balance; index++) {
+            tokenIds[index] = tokenOfOwnerByIndex(_owner, index);
+        }
+        return tokenIds;
     }
-        
+
     function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
```
Comment (UmairJanjua6) [inline contracts/VLAND.sol:53]: If we are going to use the token ID in tokenURI then we just have to concatenate the token ID with the base URI, no need to write the extra code

File: test/VLAND.test.js [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+            const minterRole = await vland.MINTER_ROLE();        
+            const expectedError = `AccessControl: account ${satoshiAddress.toLowerCase()} is missing role ${adminRole.toLowerCase()} -- Reason given: AccessControl: account ${satoshiAddress.toLowerCase()} is missing role ${adminRole.toLowerCase()}.`
+    
+            await expectRevert(
+                vland.grantRole(minterRole, satoshiAddress, { from: satoshiAddress }),
+                expectedError
+            );
+        });
+    
+        it('revokeRole emits an event if successful', async () => {
```
Comment (umairkhalid6) [inline test/VLAND.test.js:91]: poping up the error of invalid address

**Review by umairkhalid6 — CHANGES_REQUESTED** [review]
owner tokens function is working fine now. But there is some [issue](https://ibb.co/cNqJGP5) in test cases of assigning minter, admin roles (working perfect on-chain).

**Comment by jmulq** [comment]
> owner tokens function is working fine now. But there is some [issue](https://ibb.co/cNqJGP5) in test cases of assigning minter, admin roles (working perfect on-chain).

@UmairJanjua6 This is working fine for me when I run the tests. Might be worth us hopping on a screen share to see what steps you are taking to get this issue.

**Review by umairkhalid6 — APPROVED** [review]
PR reviewed by Umair. The smart contract and unit tests are working fine.

**Pull request merged**